cdk-infra/utils/ssm.ts

```diff
@@ -0,0 +1,128 @@
+import { GetParameterCommand, SSMClient } from '@aws-sdk/client-ssm';
+
+export function getSsmPathPrefix(env: string): string {
+  return `/env/${env}/docs/worker_pool`;
+}
+
+/**
+ * Returns the secure strings from SSM using the SSM client.
+ * @param ssmPrefix the path prefix that should contain the environment for the SSM strings (e.g. /env/dotcomstg/worker_pool/)
+ * @returns The map of environment variables.
+ */
+async function getSecureStrings(
+  ssmPrefix: string,
+  secureStrings: readonly string[],
+  paramToEnvMap: Map<string, string>,
+  resourceName: string
+) {
+  const ssmClient = new SSMClient({ region: process.env.CDK_DEFAULT_REGION });
+
+  const secureStringsMap: Record<string, string> = {};
+
+  await Promise.all(
+    secureStrings.map(async (paramName: string) => {
+      const getParamCommand = new GetParameterCommand({
+        Name: `${ssmPrefix}${paramName}`,
+        WithDecryption: true,
+      });
+
+      const ssmResponse = await ssmClient.send(getParamCommand);
+      const secureString = ssmResponse.Parameter?.Value;
+
+      if (!secureString) {
+        console.error(`ERROR! Could not retrieve string for the following param: ${paramName}`);
+        return;
+      }
+
+      const envName = paramToEnvMap.get(paramName);
+
+      if (!envName) {
+        console.error(
+          `ERROR! The param '${paramName}' does not having a mapping to an environment variable name. Please define this in the ${resourceName} map.`
+        );
+        return;
+      }
+
+      secureStringsMap[envName] = secureString;
+    })
+  );
+
+  return secureStringsMap;
+}
+
+const workerSecureStrings = [
+  '/npm/auth',
+  '/github/webhook/secret',
+  '/github/bot/password',
+  '/atlas/password',
+  '/fastly/docs/dochub/token',
+  '/npm/auth',
+  '/fastly/docs/dochub/service_id',
+  '/fastly/dochub_map',
+  '/fastly/docs/main/token',
+  '/fastly/docs/main/service_id',
+  '/fastly/docs/cloudmanager/token',
+  '/fastly/docs/cloudmanager/service_id',
+  '/fastly/docs/atlas/token',
+  '/fastly/docs/atlas/service_id',
+  '/fastly/docs/opsmanager/token',
+  '/fastly/docs/opsmanager/service_id',
+  '/cdn/client/id',
+  '/cdn/client/secret',
+] as const;
+
+type WorkerSecureString = typeof workerSecureStrings[number];
+
+const workerParamPathToEnvName = new Map<WorkerSecureString, string>();
+
+workerParamPathToEnvName.set('/npm/auth', 'NPM_BASE_64_AUTH');
```

Comment:
In hindsight, I am not entirely sure why I went with a `Map` object as opposed to a regular object, but oh well haha!